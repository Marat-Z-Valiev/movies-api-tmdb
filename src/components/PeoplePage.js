import React, {useState, useEffect} from "react";
import axios from "axios";
import PeopleContainer from "./PeopleContainer";
import Spinner from "./Spinner";

const PeoplePage = () => {
	let [popularPeople, setPopularPeople] = useState([]);
	let [isLoading, setIsLoading] = useState(true);

	const getPopularPeople = async () => {
		await axios
			.get(
				"https://api.themoviedb.org/3/person/popular?api_key=f8efee7e451d2ca98ae50114ad74aeeb&language=en-US&page=1"
			)
			.then((response) => {
				setPopularPeople((popularPeople = response.data.results));
				setIsLoading((isLoading = false));
			})
			.catch((err) => console.log(`this is error ${err}`));
	};

	useEffect(() => {
		getPopularPeople();
	}, []);

	return (
		<>
			<h1 style={{textAlign: "center", color: "#ffffff"}}>People</h1>
			{isLoading ? <Spinner /> : <PeopleContainer people={popularPeople} />}
		</>
	);
};

export default PeoplePage;
