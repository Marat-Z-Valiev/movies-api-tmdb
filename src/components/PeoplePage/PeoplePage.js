import React, {useEffect} from "react";
import {useDispatch, useSelector} from "react-redux";
import {
	fetchPopularPeople,
	popularPeopleSelector,
} from "../../store/slices/popularPeople";
import PeopleContainer from "../PeopleContainer/PeopleContainer";
import Spinner from "../Spinner/Spinner";
import Error from "../Error/Error";

const PeoplePage = () => {
	const dispatch = useDispatch();
	const {loading, popularPeople, hasErrors} = useSelector(
		popularPeopleSelector
	);
	useEffect(() => {
		dispatch(fetchPopularPeople());
	}, [dispatch]);

	return (
		<>
			{hasErrors ? <Error /> : ""}
			<h1 style={{textAlign: "center", color: "#ffffff"}}>People</h1>
			{loading ? <Spinner /> : <PeopleContainer people={popularPeople} />}
		</>
	);
};

export default PeoplePage;
