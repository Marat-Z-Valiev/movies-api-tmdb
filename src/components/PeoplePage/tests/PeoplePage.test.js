import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PeoplePage from "../PeoplePage";

configure({adapter: new Adapter()});

describe("PeoplePage", () => {
	it("should render correctly", () => {
		const wrapper = shallow(<PeoplePage />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
