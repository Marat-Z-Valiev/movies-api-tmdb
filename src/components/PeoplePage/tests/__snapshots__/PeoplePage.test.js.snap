// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePage should render correctly 1`] = `
<Fragment>
  <h1
    style={
      Object {
        "color": "#ffffff",
        "textAlign": "center",
      }
    }
  >
    People
  </h1>
  <Spinner />
</Fragment>
`;
